### DISCUSSION TITLE
Get rim light removed?

### DISCUSSION THREAD

**Discussion by HSDSZ** [bnpr/Malt#252] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/36077492/154784680-dacfa072-72db-4082-b28d-0a6c16a4370d.png)

The original get rim light node is not in the preview version. Anyway to achieve the same function?

**Comment by eballeste** [comment, marked as answer]
have you seen the [sample files](https://github.com/bnpr/Malt/discussions/94)? it includes an example of the `rim light` node which can be accessed by:

functions > shading - shading models > rim light

it also does not require the conversion node you are using in your screenshot and can be connected directly into the color channel of the `main pass pixel shader` output node.

![Screenshot 2022-02-19 055257](https://user-images.githubusercontent.com/6924108/154797965-916e61c9-5ef2-48fa-994d-fc1fb2b68394.png)

**Reply by HSDSZ** [reply]
Oh, I see. the "get rim light" function = "rim light" function + "surface normal" input.

**Reply by HSDSZ** [reply]
Now even without surface normal connected, "rim light" node also works

**Reply by pragma37** [reply]
The shader API has been heavily redesigned for 1.0.
If you don't find something you can use the Search operator (F3) in the node editor.
